roles/servicetelemetry/tasks/component_grafana.yml

```diff
@@ -47,94 +26,247 @@
-        definition:
-          apiVersion: integreatly.org/v1alpha1
-          kind: GrafanaDashboard
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: rhos-cloud-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: rhos-cloud-dashboard.json
-            plugins:
-              - name: grafana-polystat-panel
-                version: "1.2.11"
-            json: |
-              {{ lookup('file', 'rhos-cloud-dashboard.json') | string }}
-
-    - name: Load Infrastructure Overview Dashboard
-      k8s:
-        state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
-        definition:
-          apiVersion: integreatly.org/v1alpha1
-          kind: GrafanaDashboard
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: rhos-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: rhos-dashboard.json
-            json: |
-              {{ lookup('file', 'rhos-dashboard.json') | string }}
-
-    - name: Load Memcached Dashboard
-      k8s:
-        state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
-        definition:
-          apiVersion: integreatly.org/v1alpha1
-          kind: GrafanaDashboard
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: memcached-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: memcached-dashboard.json
-            json: |
-              {{ lookup('file', 'memcached-dashboard.json') | string }}
-
-    - name: Load Virtual Machine View Dashboard
-      k8s:
-        state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
-        definition:
-          apiVersion: integreatly.org/v1alpha1
-          kind: GrafanaDashboard
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: virtual-machine-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: virtual-machine-view.json
-            json: |
-              {{ lookup('file', 'virtual-machine-view.json') | string }}
+#---- deploy Grafana with v5 Operator (preferred)
+    - when: has_grafana_integreatly_api
+      name: Deploying with Grafana Operator v5
+      block:
+        - name: Construct oauth redirect reference
+          set_fact:
+            grafana_oauth_redir_ref:
+              kind: OAuthRedirectReference
+              apiVersion: v1
+              reference:
+                kind: Route
```

Comment:
The route name is now slightly dynamic and will change from the previously static `grafana-route` to now include the name of the STF deployment name (`default` by.... default). Tracking documentation updates at https://issues.redhat.com/browse/STF-1680